examples/documentation/cuda_overview_documentation.cu

```diff
@@ -0,0 +1,90 @@
+//  Copyright (c) 2024 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <pika/cuda.hpp>
+#include <pika/execution.hpp>
+#include <pika/init.hpp>
+
+#include <fmt/printf.h>
+#include <whip.hpp>
+
+#include <iostream>
+#include <utility>
+
+#if defined(PIKA_HAVE_CUDA)
+# include <cublas_v2.h>
+using blas_handle_t = cublasHandle_t;
+auto* blas_gemm = &cublasDgemm;
+auto blas_pointer_mode = CUBLAS_POINTER_MODE_HOST;
+auto blas_op_n = CUBLAS_OP_N;
+#elif defined(PIKA_HAVE_HIP)
+# include <rocblas/rocblas.h>
+using blas_handle_t = hipblasHandle_t;
+auto* blas_gemm = &rocblas_dgemm;
+auto blas_pointer_mode = ROCBLAS_POINTER_MODE_HOST;
+auto blas_op_n = rocblas_operation_none;
+#endif
+
+__global__ void kernel() { printf("Hello from kernel! threadIdx.x: %d\n", threadIdx.x); }
+
+int main(int argc, char* argv[])
+{
+    namespace cu = pika::cuda::experimental;
+    namespace ex = pika::execution::experimental;
+    namespace tt = pika::this_thread::experimental;
+
+    pika::start(argc, argv);
+    ex::thread_pool_scheduler cpu_sched{};
+
+    // Create a pool of CUDA streams and cuBLAS/SOLVER handles, and a scheduler that uses the pool.
+    cu::cuda_pool pool{};
+    cu::cuda_scheduler cuda_sched{pool};
+
+    {
+        // Enable polling of CUDA events on the default pool. This is required to allow the adaptors
+        // below to signal completion of kernels.
+        cu::enable_user_polling p{};
+
+        constexpr std::size_t n = 2048;
+        double* a = nullptr;
+        double* b = nullptr;
+        double* c = nullptr;
+        double alpha = 1.0;
+        double beta = 1.0;
+        whip::malloc(&a, sizeof(double) * n * n);
+        whip::malloc(&b, sizeof(double) * n * n);
+        whip::malloc(&c, sizeof(double) * n * n);
```

Comment:
I should probably initialize these with something reasonable... same in the other examples.